docs/proposals/04-etcd-member-custom-resource.md

```diff
@@ -0,0 +1,545 @@
+
+## Terminology
+
+* **druid:** [etcd-druid](https://github.com/gardener/etcd-druid) - an operator for etcd clusters.
+
+* **etcd-member:** A single etcd pod in an etcd cluster that is realised as a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/).
+
+* **backup-sidecar:** It is the etcd-backup-restore sidecar container in each etcd-member pod. 
+  
+  > **NOTE:** Term sidecar can now be confused with the latest definition in [KEP-73](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md). etcd-backup-restore container is currently not set as an `init-container` as proposed in the KEP but as a regular container in a multi-container [Pod]([Pods | Kubernetes](https://kubernetes.io/docs/concepts/workloads/pods/)).
+
+* **leading-backup-sidecar:** A backup-sidecar that is associated to an etcd leader.
+
+* **restoration:** It refers to an individual etcd-member restoring etcd data from an existing backup (comprising of full and delta snapshots). The authors have deliberately chosen to distinguish between restoration and learning. Learning refers to a process where a [learner](https://etcd.io/docs/v3.3/learning/learner/#raft-learner) "learns" from an etcd-cluster leader.
+
+## Motivation
+
+Sharing state of an individual etcd-member with druid is essential for diagnostics, monitoring, cluster-wide-operations and potential remediation. At present, only a subset of etcd-member state is shared with druid using [leases](https://kubernetes.io/docs/concepts/architecture/leases/). It was always meant as a stopgap arrangement as mentioned in the [corresponding issue](https://github.com/gardener/etcd-druid/pull/207) and is not the best use of leases.
+
+There is a need to have a clear distinction between an etcd-member state and etcd cluster state since most of an etcd cluster state is often derived by looking at individual etcd-member states. In addition, actors which update each of these states should be clearly identified so as to prevent multiple actors updating a single resource holding the state of either an etcd cluster or an etcd-member. As a consequence, etcd-members should not directly update the `Etcd` resource status and would therefore need a new custom resource allowing each member to publish detailed information about its latest state.
+
+### Goals
+
+* Introduce `EtcdMember` custom resource.
+
+* Enhance [etcd-backup-restore](https://github.com/gardener/etcd-backup-restore) to publish member specific information by updating `EtcdMember` resource. 
+
+* Today [leases](https://kubernetes.io/docs/concepts/architecture/leases/) are misused to share member-specific information with etcd-druid. Their usage to share member state [leader, follower, learner], member-id, snapshot revisions etc should be removed.
```

Comment:
Yes. The proposal wishes to eliminate the use of snapshot leases only, since that was not used in the right way. Member leases will still remain, but the annotation on it for peer-tls-enabled will go away and will be moved to `EtcdMember.Status` instead.